src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -684,14 +705,19 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
       def varsWithoutSetters = defnTrees.iterator.map(_.symbol).filter(t => t.isVar && !isExisting(t.setter))
       varsWithoutSetters.filter(v => !isSuppressed(v) && !setVars(v) && !isUnusedTerm(v))
     }
-    def unusedParams = params.iterator.filter(isUnusedParam)
+    def unusedParams = params.iterator.filter(p => !ignore(p) && isUnusedParam(p))
```

Comment:
`!ignore(p)` is redundant?

Reply:
that was "better safe than sorry", but I'll take another look.

Reply:
L712